civo/resource_database.go

```diff
@@ -0,0 +1,243 @@
+				ForceNew:     true,
+				ValidateFunc: validation.NoZeroValues,
+				Description:  "Name of the database",
+			},
+
+			"nodes": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ValidateFunc: validation.NoZeroValues,
+				Description:  "Count of nodes",
+			},
+
+			"size": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.NoZeroValues,
+				Description:  "Size of the database",
+			},
+
+			"network_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The id of the associated network",
+			},
+
+			"firewall_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The ID of the firewall to use, from the current list. If left blank or not sent, the default firewall will be used (open to all)",
+			},
+
+			"region": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The region where the database will be created.",
+			},
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+	}
+}
+
+// function to create a database
+func resourceDatabaseCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	apiClient := m.(*civogo.Client)
+
+	// overwrite the region if it is defined in the datasource
+	if region, ok := d.GetOk("region"); ok {
+		apiClient.Region = region.(string)
+	}
+
+	log.Printf("[INFO] configuring the database %s", d.Get("name").(string))
+
+	config := &civogo.CreateDatabaseRequest{
+		Name:   d.Get("name").(string),
+		Region: apiClient.Region,
+	}
+
+	if attr, ok := d.GetOk("nodes"); ok {
+		config.Nodes = attr.(int)
+	}
+
+	if attr, ok := d.GetOk("size"); ok {
+		config.Size = attr.(string)
+	}
+
+	if networtID, ok := d.GetOk("network_id"); ok {
+		config.NetworkID = networtID.(string)
+	} else {
+		defaultNetwork, err := apiClient.GetDefaultNetwork()
+		if err != nil {
+			return diag.Errorf("[ERR] failed to get the default network: %s", err)
+		}
+		config.NetworkID = defaultNetwork.ID
+	}
+
+	if attr, ok := d.GetOk("firewall_id"); ok {
+		firewallID := attr.(string)
+		firewall, err := apiClient.FindFirewall(firewallID)
+		if err != nil {
+			return diag.Errorf("[ERR] unable to find firewall - %s", err)
+		}
+
+		if firewall.NetworkID != config.NetworkID {
+			return diag.Errorf("[ERR] firewall %s is not part of network %s", firewall.ID, config.NetworkID)
+		}
+
+		config.FirewallID = firewallID
+	}
+
+	log.Printf("[INFO] creating the Database %s", d.Get("name").(string))
+	database, err := apiClient.NewDatabase(config)
+	if err != nil {
+		return diag.Errorf("[ERR] failed to create Database: %s", err)
+	}
+
+	d.SetId(database.ID)
+
+	createStateConf := &resource.StateChangeConf{
+		Pending: []string{"pending"},
+		Target:  []string{"ready"},
```

Comment:
`pending` and `ready` are the state of the database ??

Reply:
Yes, but I want a confirmation if "pending" and "ready"are correct for database or we have different state for database.

Reply:
@vishalanarase can you confirm this ??

Reply:
Yes that is correct

Reply:
@satakshigarg change it to `Pending` and `Ready`

Reply:
Done !!